### ISSUE TITLE
Refactoring PyTorch version-specific attributes to aid experimentation

### ISSUE THREAD

**Issue by etetteh** [ivadomed/ivadomed#1095]
I have some few out-of-scope issues about library versions, and I don’t know if the time is right to comment on that. I understand ivadomed currently uses `torch==1.8.0`, but some issues arise when experimenting with ideas based on higher versions of `torch`.

My suggestion would be to refactor some parts (will highlight later) of `ivadomed`, so that experimenting with new ideas becomes seamless. For instance,
1. [this issue](https://github.com/ivadomed/ivadomed/issues/1094) about importing int_classes  from `torch._six` may be resolved by removing the import, and simply replacing `int` where `int_classes` need to be.
2. Some LR schedulers or some LR schedulers of some versions of `torch` does not have the attribute `.get_last_lr()` (as used [here](https://github.com/ivadomed/ivadomed/blob/d4181999c7dcda3d0ef22161eaf3e32bf6d92b9f/ivadomed/training.py#L156)), but this may be resolved by replacing `scheduler.get_last_lr()[0]` with `[group['lr'] for group in optimizer.param_groups][0] `.

Basically, what I am saying is that sticking to the current `torch==1.8.0` version of ivadomed limits experimentation with ideas built with higher versions, and one solution is to refactor ivadomed so that it doesn’t strictly rely on version-specific attributes/methods, at least for now.

**Comment by dyt811** [comment]
Thanks @etetteh, I think these would be great initiatives. 

What I would recommend is to carry the following: 

1) Like what you have done [here](https://github.com/ivadomed/ivadomed/pull/1093#discussion_r820894987), either make the PR yourself or let us know the locations to make these necessary changes such as what you have done in #1094, and we get get those PRs made and you can review and approve them. 

2) In the mean time, in parallel, you can continue experimentation on your branch with all those above changes, and as the main branch gradually catches up to your changes, you can simply merge from master to your experimental branch and resolve these minor conflicts since it will be identical to your branch, the resolution will be easy. 

We have found this approach offers the best trade off of main branch stability as well as allow experimentation. 

The key part is to ensure all the necessary changes you want happen, gets described, documented in issued, and PRed (either by you, usually faster; or by explaining to the dev team and made by them and you review it, which is slower) into the main branch for long term stability support. A key things to keep in mind is that ivadomed currently have no multiple full time devs (I am part time) dedicated to it so the pace of the development is very priority driven and the master is not updated enough to allow experimentation on a "rapid prototype" level. Hopefully with your help, we will get closer to that goals though as you are more in tune with the latest feature development. So please keep the good ideas coming and we ill incorporate them as soon as we can. 

BTW, I have added your issue to the meeting minute to bring greater awareness to the above problems. We may  need a second issue for point 2 for PR purposes but please feel free to update and add more.

**Comment by etetteh** [comment]
Thank you @dyt811. 
After careful analysis of the changes of the torch versions, it seems the issue with `get_last_lr()` attribute has been fixed from version `1.9.1` to the latest version of `1.11.0`.
The other issue with importing `int_classes` from `torch._six` is the only existing issue for now.
Please, find the changes in my latest PR or check it [here](https://github.com/etetteh/ivadomed/tree/et/version_invariant_attributes)

**Comment by dyt811** [comment]
PR is now merged. @etetteh can you close this issue if everything is satisfactory, or issue anything new with regard to adopting standard to help with future versions of Torch.

**Comment by etetteh** [comment]
Everything is satisfactory for now. closing this issue.
